mui/mui.py

```diff
@@ -0,0 +1,119 @@
+import tempfile
+from typing import Set
+
+from binaryninja import (
+    PluginCommand,
+    BinaryView,
+    BackgroundTaskThread,
+    HighlightStandardColor,
+    HighlightColor,
+    show_message_box,
+    MessageBoxButtonSet,
+    MessageBoxIcon,
+)
+from manticore.core.state import StateBase
+from manticore.native import Manticore
+
+BinaryView.set_default_session_data("mui_find", set())
+BinaryView.set_default_session_data("mui_avoid", set())
+
+
+class ManticoreRunner(BackgroundTaskThread):
+    def __init__(self, find: Set[int], avoid: Set[int], view: BinaryView):
+        BackgroundTaskThread.__init__(self, "Solving with manticore...", True)
+        self.find = tuple(find)
+        self.avoid = tuple(avoid)
```

Comment:
Good catch for making these collections immutable when handling them with Manticore 👍  💯